src/post/post.service.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { User } from 'src/user/user.interface';
+import { createPostDto, deletePostDto, updatePostDto } from './dto';
+import { Post } from './post.interface';
+
+@Injectable()
+export class PostService {
+  constructor(
+    @InjectModel('Post') private readonly postModel: Model<Post>,
+    @InjectModel('User') private readonly userModel: Model<User>,
+  ) {}
+  async createPost(id: string, dto: createPostDto) {
```

Comment:
Function return types should be defined as well to fulfil our use of TypeScript.